src/repositories/UniswapV3PoolRepository.ts

```diff
@@ -0,0 +1,46 @@
+import {injectable} from 'inversify';
+import {getModelForClass} from '@typegoose/typegoose';
+
+import {UniswapV3Pool} from '../models/Pools.js';
+import {ChainsIds} from '../types/ChainsIds.js';
+
+@injectable()
+export class UniswapV3PoolRepository {
+  async save(
+    props: Omit<
```

Comment:
why we need such a weird construction? if we need more than one `save` method, let's create them.